### PR TITLE
feat Add container id validate

### PR DESCRIPTION
**Pull request by lengrongfu**
issue: https://github.com/containers/youki/issues/1580

### CODE REVIEW

File: crates/libcontainer/src/container/builder.rs [hunk 1]
```diff
@@ -37,6 +37,9 @@ pub struct ContainerBuilder<'a> {
 /// .build();
 /// ```
 impl<'a> ContainerBuilder<'a> {
+    /// ERR_INVALID_ID is container id validate error
+    const ERR_INVALID_ID: &str = "invalid container ID format";
```
Comment (yihuaf) [inline crates/libcontainer/src/container/builder.rs:41]: Can you please use the `anyhow` error and result, following how the rest of the code is doing? This is actually a good example where we should use `thiserror`, but we have not done this yet.

File: crates/libcontainer/src/container/builder.rs [hunk 2]
```diff
@@ -50,7 +53,7 @@ impl<'a> ContainerBuilder<'a> {
     /// ```
     pub fn new(container_id: String, syscall: &'a dyn Syscall) -> Self {
         let root_path = PathBuf::from("/run/youki");
-
+        Self::validate_id(container_id.clone()).unwrap();
```
Comment (yihuaf) [inline crates/libcontainer/src/container/builder.rs:56]: I like the idea of validating the ID, but perhaps this check should happen before getting to the builder.

File: crates/libcontainer/src/container/builder.rs [hunk 3]
```diff
@@ -225,6 +261,27 @@ mod tests {
             .context("build container")?;
         assert_eq!(path_builder.pid_file, Some(cwd.join("not/existing/path")));
 
+        // validate container_id
+        let result =
```
Comment (yihuaf) [inline crates/libcontainer/src/container/builder.rs:265]: These tests should be in their own testcases.

File: crates/libcontainer/src/container/builder.rs [hunk 4]
```diff
@@ -61,6 +64,39 @@ impl<'a> ContainerBuilder<'a> {
...
+    /// - uppercase (A-Z) and lowercase (a-z) Latin letters;
+    /// - digits (0-9);
+    /// - underscore (_);
+    /// - plus sign (+);
+    /// - minus sign (-);
+    /// - period (.).
+    ///
+    /// In addition, IDs that can't be used to represent a file name
+    /// (such as . or ..) are rejected.
+    fn validate_id(container_id: String) -> Result<(), &'static str> {
```
Comment (yihuaf) [inline crates/libcontainer/src/container/builder.rs:84]: This is good.

File: crates/libcontainer/src/container/builder.rs (hunk 1 above)
Comment (lengrongfu) [reply crates/libcontainer/src/container/builder.rs]: Already using `anyhow` error and Result, I can submit pr later to use `thiserror`.

File: crates/libcontainer/src/container/builder.rs (hunk 2 above)
Comment (lengrongfu) [reply crates/libcontainer/src/container/builder.rs]: Please take a look, it has been modified to perform verification before creating the builder.

File: crates/libcontainer/src/container/builder.rs (hunk 3 above)
Comment (lengrongfu) [reply crates/libcontainer/src/container/builder.rs]: done

**Review by yihuaf — APPROVED** [review]
Much better :)

**Pull request merged**